R/agg_scale.R

```diff
@@ -0,0 +1,885 @@
+#' each other. For example if aggregating or scaling locations then mapping
+#' needs to define how each child location relates to each parent location. It
+#' is then assumed that each node in the `mapping` hierarchy will need to be
+#' aggregated to.
+#'
+#' For numeric interval variables the `mapping` argument must have columns for
+#' `{col_stem}_start` and `{col_stem}_end` defining the start and end of each
+#' aggregate interval.
+#'
+#' These functions work even if `dt` is not a square dataset. Meaning it is okay
+#' if different combinations of `id_vars` have different `col_stem` values. For
+#' example if making age aggregates, it is okay if some location-years have
+#' 5-year age groups while other location-years have 1-year age groups.
+#'
+#' @examples
+#' # aggregate count data from present day Iran provinces to historical
+#' # provinces and Iran as a whole
+#' input_dt <- data.table::CJ(location = iran_mapping[!grepl("[0-9]+", child),
+#'                                                    child],
+#'                            year = 2011,
+#'                            value = 1)
+#' output_dt <- agg(dt = input_dt,
+#'                  id_cols = c("location", "year"),
+#'                  value_cols = "value",
+#'                  col_stem = "location",
+#'                  col_type = "categorical",
+#'                  mapping = iran_mapping)
+#'
+#' # scale count data from present day Iran provinces to Iran national value
+#' input_dt <- data.table::CJ(location = iran_mapping[!grepl("[0-9]+", child),
+#'                                                    child],
+#'                            year = 2011,
+#'                            value = 1)
+#' input_dt_agg <- data.table::data.table(
+#'   location = "Iran (Islamic Republic of)",
+#'   year = 2011, value = 62
+#' )
+#' input_dt <- rbind(input_dt, input_dt_agg)
+#' output_dt <- scale(dt = input_dt,
+#'                    id_cols = c("location", "year"),
+#'                    value_cols = "value",
+#'                    col_stem = "location",
+#'                    col_type = "categorical",
+#'                    mapping = iran_mapping,
+#'                    collapse_missing = TRUE)
+#'
+#' # aggregate age-specific count data
+#' input_dt <- data.table::data.table(year = 2010,
+#'                         age_start = seq(0, 95, 1),
+#'                         value1 = 1, value2 = 2)
+#' gen_end(input_dt, id_cols = c("year", "age_start"), col_stem = "age")
+#' age_mapping <- data.table::data.table(age_start = c(0, 15, 85),
+#'                                       age_end = c(5, 60, Inf))
+#' output_dt <- agg(dt = input_dt,
+#'                  id_cols = c("year", "age_start", "age_end"),
+#'                  value_cols = c("value1", "value2"),
+#'                  col_stem = "age",
+#'                  col_type = "interval",
+#'                  mapping = age_mapping)
+#'
+#' # scale age-specific probability data
+#'
+#' @export
+#' @rdname agg_scale
+agg <- function(dt,
```

Comment:
Do we have a style preference on single arg per line vs wrapping args? I tend to go back and forth on this, but generally prefer wrapping args in the function definition and single arg per line when the function is called in code or within other functions.

Reply:
Stick to these guidelines? Which recommend if it goes over one line to do single arg per line except in special cases. I tend to wrap too

https://style.tidyverse.org/syntax.html#long-lines

https://style.tidyverse.org/functions.html#long-lines-1

Reply:
Added to internal style guide.